tests/filecheck/dialects/riscv/riscv_ops_custom_syntax.mlir

```diff
@@ -0,0 +1,270 @@
+  %csrrwi_rw = "riscv.csrrwi"() {"csr" = 1024 : i32, "immediate" = 1 : i32}: () -> !riscv.reg<>
+  %csrrwi_w = "riscv.csrrwi"() {"csr" = 1024 : i32, "writeonly", "immediate" = 1 : i32}: () -> !riscv.reg<>
+  // CHECK-NEXT: %csrrw_rw = riscv.csrrw %0, 1024 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrw_w = riscv.csrrw %0, 1024, w : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrs_rw = riscv.csrrs %0, 1024 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrs_r = riscv.csrrs %0, 1024, r : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrc_rw = riscv.csrrc %0, 1024 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrc_r = riscv.csrrc %0, 1024, r : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrsi_rw = riscv.csrrsi 1024, 8 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrsi_r = riscv.csrrsi 1024, 0 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrci_rw = riscv.csrrci 1024, 8 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrci_r = riscv.csrrci 1024, 0 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrwi_rw = riscv.csrrwi 1024, 1 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %csrrwi_w = riscv.csrrwi 1024, w, 1 : () -> (!riscv.reg<>)
+
+  // Machine Mode Privileged Instructions
+  "riscv.wfi"() : () -> ()
+  // CHECK-NEXT: riscv.wfi : () -> ()
+
+
+  // RV32M/RV64M: 7 “M” Standard Extension for Integer Multiplication and Division
+
+  // Multiplication Operations
+  %mul = "riscv.mul"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulh = "riscv.mulh"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulhsu = "riscv.mulhsu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %mulhu = "riscv.mulhu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %mul = riscv.mul %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulh = riscv.mulh %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulhsu = riscv.mulhsu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %mulhu = riscv.mulhu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+
+  // Division Operations
+  %div = "riscv.div"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %divu = "riscv.divu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %rem = "riscv.rem"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %remu = "riscv.remu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %div = riscv.div %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %divu = riscv.divu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %rem = riscv.rem %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %remu = riscv.remu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+
+  // Assembler pseudo-instructions
+
+  %li = "riscv.li"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
+  // CHECK-NEXT: %li = riscv.li 1 : () -> (!riscv.reg<>)
```

Comment:
Minor nitpick: Similarly, we can simplify the type signature to `: !riscv.reg<>` since we know it will never have arguments and only ever a single return value.